### DISCUSSION TITLE
Annotators not working. No error messages, it runs but the outputs are not right

### DISCUSSION THREAD

**Discussion by sooolee** [Mikubill/sd-webui-controlnet#884] [Q&A]
Hi! It would be great if someone can share some insights. 

I updated the ControlNet version, and downloaded the models correctly (using 'downloading', not through the links) and place them in the correct folder along with the yaml files. I'm currently only using Canny, Depth, Normal, Scrabble, so downloaded only these models. 

When I run it though, the annotator outputs are rather random. It's not really reading the given images. I don't get any error messages, but the outputs (preview) are wrong. It used to work fine with the previous version. 

Any ideas about where things could have gotten wrong? Thanks in advance!

**Comment by lllyasviel** [comment]
update and restart completely

**Reply by sooolee** [reply]
Thanks for the reply! However, I did that - update the extension, and re-start from the terminal. Also deleted the extension, and re-installed several times, followed by extension update and restart. Didn't work.

One thing though. I installed Auto1111 in Linux, cloud server. I didn't restart the instance. Would that be a problem?

**Comment by revolverocelot1** [comment]
are you sure, for me its working fine. can you give the error message or its just blanked

**Comment by lllyasviel** [comment]
screenshots?

**Comment by sooolee** [comment]
extension up to date:
![image](https://user-images.githubusercontent.com/95900980/232707212-3938712a-cc5a-4183-bad1-fb83898607ac.png)

But the output is not right:
![image](https://user-images.githubusercontent.com/95900980/232707493-fac837d8-fb1d-4fe9-81d8-d322293e7492.png)

And no error message (after running the above):
![image](https://user-images.githubusercontent.com/95900980/232709032-6208cca2-fba7-42fc-bce3-7e4ad37c55dd.png)

**Comment by sooolee** [comment]
Under the following path:

![image](https://user-images.githubusercontent.com/95900980/232711467-639f2df2-0e6e-4563-99a1-e75725897255.png)


pth and yaml files: 

![image](https://user-images.githubusercontent.com/95900980/232711552-76cfaa69-fc3d-44d5-8cf2-082f6445ca97.png)

**Comment by sooolee** [comment]
But the annotator folder seems a bit messy. I didn't touch this part, but in some running, it created 'downloads' folder for leres and midas.

![image](https://user-images.githubusercontent.com/95900980/232712423-cc7521e4-e9bf-4085-8943-0620d6d2d9a0.png)

![image](https://user-images.githubusercontent.com/95900980/232712552-2947a846-23e8-42c0-927c-48673d66f308.png)

**Reply by ljleb** [reply]
Yes this folder is now expected to hold annotator model weights by default, IIUC it is unrelated to the problem you are facing.

**Comment by lllyasviel** [comment]
why is this even possible? big-endian vs little-endian linux?

**Comment by lllyasviel** [comment]
perhaps it is fixed here https://github.com/Mikubill/sd-webui-controlnet/pull/945

**Reply by sooolee** [reply]
Thanks for looking into this. Unfortunately, now I get an error message. I tried this time a simple txt2img with ControlNet. The error message is:
"RuntimeError: mat1 and mat2 shapes cannot be multiplied (154x1024 and 768x320)"

Same error message when I use img2img with ControlNet. I use Ubuntu 22.10 (GNU/Linux 5.19.0-1018-gcp x86_64).  I haven't had time to try some more recently. But I will try to see if there is anything wrong on my end. 

![image](https://user-images.githubusercontent.com/95900980/233815520-48eef67b-4d87-45a2-9e92-9698d8ef8fe4.png)

**Reply by TheOnlyHolyMoly** [reply]
Yes, I have had the same runtime error on img2img. I thought it was because of missing size match between reference image and control net image, but I havent figured it out fully.

**Comment by lllyasviel** [comment]
hello we provided option to use previous version.

See also "Use Previous Version" in the readme

**Comment by JoshuaJewell** [comment]
I was having a similar issue on one instance of Firefox and found that opening the UI on Chromium was able to fix the problem. I might look at what settings specifically are causing an issue since it runs perfectly fine with Firefox on another computer.

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/79232160/9ea604f0-17a4-4cd9-be56-7b9fde9ed436)
Here I used the inbuilt painting tool on ControlNet and it seems that despite images not entering the annotator correctly, it is still performing the correct operation but only on information introduced inside the UI.
